### PR TITLE
package.rb: Accept environment variables and options in `system` function

### PR DESCRIPTION
**Pull request by supechicken**
Tested on `x86_64`

Currently, the patched `system` function will return errors when `hash` is passed as an argument, which means all `Kernel.system` options like `chdir: ` and `exception:` are not working now.

This PR fixed it, some changes were added also:

### Changes
- `-j` argument will apply to `samu` and `ninja` now
- `CREW_ENV_OPTIONS` are passed to all commands by default (can be covered by passing a new value in cmdline/hash)
- Use `bash` instead of `/bin/sh` if a shell is needed (most situations)

### Run the following to get this pull request's changes locally for testing.
```
CREW_TESTING_REPO=https://github.com/supechicken666/chromebrew.git CREW_TESTING_BRANCH=package_rb CREW_TESTING=1 crew update
```

### CODE REVIEW

**Comment by satmandu** [comment]
Do no `samu` and `ninja` automatically use `nproc`, so they don't need to have `-j` passed to them unless you are overriding the default of "maximum parallelism?"

**Comment by supechicken** [comment]
> Do no `samu` and `ninja` automatically use `nproc`, so they don't need to have `-j` passed to them unless you are overriding the default of "maximum parallelism?"

I am not sure about that, this is the output of `ninja` on my system:
chronos@localhost /usr/local $ nproc
8
chronos@localhost /usr/local $ ninja --help   
...snip...
  -j N     run N jobs in parallel (0 means infinity) [default=10 on this system]
```
The default value of `-j` doesn't match the output of `nproc`

**Comment by satmandu** [comment]
Looks like the `ninja` default is `nproc +2` which seems like a reasonable default, so maybe we shouldn't mess with it?

File: lib/package.rb [hunk 1]
```diff
@@ -152,39 +154,53 @@ def self.system(*args)
...
+        # pass to Kernel.system directly since it doesn't need a shell
+        Kernel.system env_options,
+                      *cmd_args,      # ensure the command is passed first
+                      system_options  # pass Kernel.system options (all args after command) then
       end
+    rescue => e
+      exitstatus = $?.exitstatus
+      # print failed line number and error message
+      puts "#{e.backtrace[1]}: #{e.message}".orange
+      raise InstallError.new("`#{cmd_args.join(' ')}` exited with #{exitstatus}")
```
[the remark this answers is not in the document]
Comment (supechicken) [reply lib/package.rb:203]: Actually, it never works due to the variable scope (`@opt_verbose` is an instance variable defined in the main `Object` class and cannot be called in a different class like `Package`)

**Comment by supechicken** [comment]
> Looks like the `ninja` default is `nproc +2` which seems like a reasonable default, so maybe we shouldn't mess with it?

Removed for `ninja` and `samu`

**Comment by uberhacker** [comment]
@supechicken:  Bump version?

**Pull request merged**